Project-II-Calculator/calculator/src/App.js

```diff
@@ -1,23 +1,119 @@
 import React from 'react';
 import './App.css';
+import CalculatorDisplay from './components/DisplayComponents/CalculatorDisplay';
+import ActionButton from './components/ButtonComponents/ActionButton';
+import NumberButton from './components/ButtonComponents/NumberButton';
 
-const App = () => {
-  return (
-    <div>
-      <h3>Welcome to React Calculator</h3>
-      <p>
-        We have given you a starter project. You'll want to build out your
-        components in their respective files, remove this code and replace it
-        with the proper components.
-      </p>
-      <p>
-        <strong>
-          Don't forget to `default export` your components and import them here
-          inside of this file in order to make them work.
-        </strong>
-      </p>
-    </div>
-  );
-};
-
-export default App;
+const BUTTONS = [
+  { text: 'C', buttonStyle: 'num' },
+  { text: '7', buttonStyle: 'num' },
+  { text: '8', buttonStyle: 'num' },
+  { text: '9', buttonStyle: 'num' },
+  { text: '4', buttonStyle: 'num' },
+  { text: '5', buttonStyle: 'num' },
+  { text: '6', buttonStyle: 'num' },
+  { text: '1', buttonStyle: 'num' },
+  { text: '2', buttonStyle: 'num' },
+  { text: '3', buttonStyle: 'num' },
+  { text: '0', buttonStyle: 'num' },
+
+];
+
+const ACTIONBUTTONS = [
+  { text: '/', buttonStyle: 'action' },
+  { text: '*', buttonStyle: 'action' },
+  { text: '-', buttonStyle: 'action' },
+  { text: '+', buttonStyle: 'action' },
+  { text: '=', buttonStyle: 'action' },
+]
+
+class App extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      display: '',
+      num1: '',
+      num2: '',
+      working: false,
+      oper:'',
+      end: false,
+    };
+  }
+
+  clickHandler = event => {
+    let button = event.target.innerHTML;
+    if (this.state.num1 === '') document.querySelector('.key-eq-sign').style.background = 'maroon';
```

Comment:
I really like that you implemented this feature, but you should be able to do so without hitting the DOM with querySelector. one way to approach this would be to pass down the relevant pieces of state to your equal sign component, and inside the equal sign component have a similar conditional that gives it an extra className, which you could apply the different color to in your css